### ISSUE TITLE
Wrong error 4104, mistaking readonly as mutable.

### ISSUE THREAD

**Issue by miguel-leon** [microsoft/TypeScript#50634]
# Bug Report

The following example:
```ts
const X = factory<{ a: number[] }>();

const a: readonly number[] = [1, 2, 3];

X.foo({ a }); // error 4104, but foo expects readonly! 🤷‍♂️


function factory<T>() {
	return {
		foo(value: Readonly<T>) {
		}
	};
}
```

### 🔎 Search Terms

readonly, mutable, 4104

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?jsx=0#code/MYewdgzgLgBAGjAvDAZgQ2FEAnAngHgG8Y0AuGMAVwFsAjAU2wG0BdGAXwD4AKASgG4AsACgRoSLDRIYTAIwAaGACZFAZjZoIMcdCGjhcAHQoQIbsSnsBIm8JSUwmAJbhUGLHnwAVHrxiERAEhseihKbDB-IMCTEHwAVRh6AA8oejAAEy0AJXo0DPAAGwIfHgA3NELKenJ4vwDhQMD2IPY9FuEgA)

### 🙁 Actual behavior

Error 4104

### 🙂 Expected behavior

No error
